magicctapipe/io/io.py

```diff
@@ -172,6 +219,91 @@ def get_stereo_events(
     return event_data_stereo
 
 
+
+def get_stereo_events(
+    event_data, config, quality_cuts=None, group_index=["obs_id", "event_id"], eval_multi_combo=True
+):
+    """
+    Gets the stereo events surviving specified quality cuts.
+
+    It also adds the telescope multiplicity `multiplicity` and
+    combination types `combo_type` to the output data frame.
+
+    Parameters
+    ----------
+    event_data: pandas.core.frame.DataFrame
+        Data frame of shower events
+    config: dict 
+        Read from the yaml file with information about the telescope IDs. Typically called "config_general.yaml"
+    quality_cuts: str
+        Quality cuts applied to the input data
+    group_index: list
+        Index to group telescope events
+    
+
+    Returns
+    -------
+    event_data_stereo: pandas.core.frame.DataFrame
+        Data frame of the stereo events surviving the quality cuts
+    """
+    
+    TEL_NAMES, TEL_COMBINATIONS = telescope_combinations(config)
+    
+    event_data_stereo = event_data.copy()
+
+    # Apply the quality cuts
+    if quality_cuts is not None:
+        event_data_stereo.query(quality_cuts, inplace=True)
+    
+    max_multiplicity=len(TEL_NAMES.keys())
+    # Extract stereo events
+    event_data_stereo["multiplicity"] = event_data_stereo.groupby(group_index).size()
+    event_data_stereo.query(f"multiplicity >1 & multiplicity <= {max_multiplicity}", inplace=True)
```

Comment:
why do you need to query for maximum multiplicity? since it is defined as all possible telescopes, this should be always fulfilled

Reply:
While testing these scripts a few months ago, we found some (<<1%) LST sub-runs in which, after the coincidence step, for ~10 events (or even less) the multiplicity was evaluated as 4 (with 3 telescopes!!!!). I also asked about hem to Ruben, but we could not find out the causes (maybe some corrupted events, or duplicated events, or some Inf/nan values causing a failure...). So these events must not be used (BTW, they cause the scripts to crash, so...).
 In the master we use "multiplicity == [2, 3]" so we never run into this case, but with a number of telescopes >(=)2 and <=6 we cannot fix a list of multiplicity values as in the master, but obviously the multiplicity (i.e., number of telescopes detecting an event) cannot be greater than the size of the array used!
Obviously, if you know the cause, we can try to fix the data instead of using this 'quality cut' to discard anomalous events

Reply:
can you point me to example of data with this problem? I was actually also checking a few times if there are no strange multiplicities, and never saw anything beyond 3 (might have been removed by the original cut however, this I'm not sure anymore at which point I was checking it. With a concrete example I can have a look.

The issue is that if you have such a problem that adds an extra telescope to end up with 4. very likely it would add also an extra telescope to 2 telescope events making a fake 3 tel event that will survive the analysis.

Reply:
I have to check it again (we implemented this cut more than 6 months ago...). I'm running again a Crab analysis with the same dataset and the current pipeline to check if I can reproduce the same error (w/o multiplicity limit in get_stereo). If not, we can remove this limit, otherwise we have to check the data. I'll let you know as soon as possible

Reply:
I tried to remove this limit and, on the complete Crab paper dataset, I run into no error. Actually, we added this limit in May and, in the meanwhile, the pipeline has been upgraded (but I never tried to remove it), sorry, my fault. BTW, I will remove it now as no longer needed